TSOClient/FSO.Windows/Program.cs

```diff
@@ -32,6 +32,8 @@ public static void Main(string[] args)
                 var startProxy = new GameStartProxy();
                 startProxy.Start(UseDX);
             }
+
+            Environment.Exit(0);
```

Comment:
Monogame is meant to call `TSOGame.OnExiting`, but for some reason it isn't doing that. Maybe they broke the event at some point.

Without the event, the `RCWorkerLoop` threads never terminate.

Reply:
The archive client (and simitone) also have some code that prevents the game from being closed that isn’t working anymore. Simitone uses it to show the save confirmation dialog, and the archive client uses it to safely shut down the server.

I’ll have a look at why - it does seem to add the closing event to the form, but setting e.Cancel doesn’t stop it from closing. Maybe these issues are connected.

Reply:
What about Environment.Exit(0); it is usable?